### ISSUE TITLE
How to set up different forward for train/val/test mode?

### ISSUE THREAD

**Issue by dxu312** [facebookresearch/mmf#889]
Hi,
I'd like to add some weighting to the outputs of the models for the training, but not during validation and testing. Reading all instructions/codes, so far I can't figure out how I can separate like that in MMF. Would you please advice if there's a way? Thanks a lot!

Best,
Da

**Comment by apsdehal** [comment]
Hi,
SampleList that gets passed to the model has `dataset_type` attribute which can have value `train|val|test` and you can use that to make decisions based on `if sample_list.dataset_type == "train"` to only do something during training mode. 

Alternatively, when the model is running on validation and testing data, the model is in eval mode so you can check `self.training` to know whether you are in training mode or not.
